File: TypeScript/regex-bench.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fs = require("fs");
var xml = require("xml2js");
var data = fs.readFileSync("../parser_data/101301020-show-version.txt", "ascii").toString().split('\r\n');
var catelogData = fs.readFileSync("../parser_data/discovery_parser.xml", "ascii");
var totalRegexs = 0;
xml.parseString(catelogData, function (err, result) {
    for (var _i = 0, _a = result.parserCollection.commandOutputParser; _i < _a.length; _i++) {
        var cmdParser = _a[_i];
        for (var _b = 0, _c = cmdParser.parser; _b < _c.length; _b++) {
            var parse = _c[_b];
            if (parse.$['start']) {
                var reg = new RegExp(parse.$['start'], "i");
                totalRegexs++;
                for (var i = 0; i < data.length; i++) {
                    if (reg.test(data[i])) {
                        console.log(data[i]);
                    }
                }
            }
        }
    }
    console.log("\n\nRan " + totalRegexs + " regex searches");
});

File: go/regex-bench.go
package main

import (
	"encoding/xml"
	"fmt"
	"io/ioutil"
	"regexp"
	"strings"
)

type parserCollection struct {
	XMLName             xml.Name      `xml:"parserCollection"`
	Version             string        `xml:"xsi,attr"`
	CommandOutputParser []outputParse `xml:"commandOutputParser"`
}

type outputParse struct {
	XMLName xml.Name `xml:"commandOutputParser"`
	Name    string   `xml:"name,attr"`
	Parser  []parse  `xml:"parser"`
}

type parse struct {
	XMLName     xml.Name `xml:"parser"`
	Start       string   `xml:"start,attr"`
	Description string   `xml:",innerxml"`
}

func main() {
	totalRegexs := 0
	xmlData, _ := ioutil.ReadFile("../parser_data/discovery_parser.xml")
	inp, _ := ioutil.ReadFile("../parser_data/101301020-show-version.txt")
	splitInp := strings.Split(string(inp), "\n")

	parseXml := parserCollection{}
	xml.Unmarshal(xmlData, &parseXml)

	for _, outParse := range parseXml.CommandOutputParser {
		for _, parser := range outParse.Parser {
			if parser.Start != "" {
				totalRegexs++

				for indx, _ := range splitInp {
					matched, _ := regexp.MatchString(parser.Start, splitInp[indx])
					if matched {
						fmt.Println(splitInp[indx])
					}
				}
			}
		}
	}

	fmt.Printf("\n\nRan %d regex searches", totalRegexs)
}
